lua/lean/init.lua

```diff
@@ -73,4 +70,36 @@ function lean.use_suggested_mappings(buffer_local)
   end
 end
 
+--- Is the current buffer a lean buffer?
+function lean.is_lean_buffer()
+  return vim.bo.ft == "lean" or vim.bo.ft == "lean3"
+end
+
+--- Return the current Lean search path.
+---
+--- Includes both the Lean core libraries as well as project-specific
+--- directories.
+function lean.current_search_paths()
+  local paths
+
+  if vim.bo.ft == "lean3" then
+    paths = require'lean.lean3'.__current_search_paths()
+  else
+    local root = vim.lsp.buf.list_workspace_folders()[1]
+    paths = vim.tbl_map(
+      function(path) return root .. '/' .. path end,
+      subprocess_check_output({"leanpkg", "print-paths"}, { cwd = root })
+    )
+    vim.list_extend(
+      paths,
+      subprocess_check_output({"lean", "--print-libdir"}, { cwd = root })
```

Comment:
This I think could add a lot of noise to the search - perhaps we should make it configurable whether to include the standard library?

Reply:
We can also more generally start thinking about how to allow someone to choose what dependencies are included in the search (after splitting as suggested by Gabriel above)

Reply:
Actually, I have an idea for some upstream changes to `telescope.nvim`, which should allow for this and also make it so that we can configure integration here (rather then asking users to do this in their dotfiles). Will make an issue with more details tomorrow.

Reply:
I guess it depends how much overlap what you type has with stuff in the stdlib -- I could certainly believe it'd be useful to not show it though. But yeah maybe in a followup? There's other stuff I want to hit.

Reply:
Searching the stdlib is the only reason why I think this feature should be in lean.nvim at all.  I already have `<leader>fg` set up to do live grep in any project, Lean or not (this is a one-liner directly copied from the telescope.nvim readme).

In mathlib, the boundary between what is in the core library and what is in mathlib is not very clearly defined.  Some basic definitions are in mathlib, others in core.  So if you're already searching through all of mathlib, you should probably search through the core library as well.